### PR TITLE
MBL-224 cross server traffic

### PR DESCRIPTION
**Pull request by joanxie878**
What this PR do?

We want to add means to generate cross server so that we can send data from non-authority server to other server, we use the worker flag "cross_server_size" to set the size of the array data we need to send, and we use the worker flag "cross_server_frequency" to determine how many seconds we send data once.

Test Link:
https://buildkite.com/improbable/unrealgdk-nfr/builds/4255 (2_1 zone for 1size*5frequency)

Test Result:
image

### CODE REVIEW

File: Game/Source/GDKTestGyms/AUptimeGameMode.cpp [hunk 1]
```diff
@@ -391,4 +416,59 @@ AActor* AUptimeGameMode::FindPlayerStart_Implementation(AController* Player, con
...
+	TArray<FVector> Locations = GenerateCrossServerLoaction();
+	auto SizeOfLocations = Locations.Num();
+	for (auto i = 0; i < SizeOfLocations; ++i)
+	{
+		AUptimeCrossServerBeacon* Beacon = World->SpawnActor<AUptimeCrossServerBeacon>(CrossServerClass, Locations[i], FRotator::ZeroRotator, FActorSpawnParameters());
+		checkf(Beacon, TEXT("Beacon failed to spawn at %s"), *Locations[i].ToString());
+
+		UDeterministicBlackboardValues* Comp = Cast<UDeterministicBlackboardValues>(Beacon->FindComponentByClass(UDeterministicBlackboardValues::StaticClass()));
+		checkf(Comp, TEXT("Beacon must have a UDeterministicBlackboardValues component."));
+		Comp->ClientSetBlackboardAILocations(CrossServerRunPoints[i]);
```
Comment (jayprobable) [inline Game/Source/GDKTestGyms/AUptimeGameMode.cpp:439]: Why do the cross server beacons have run points?

File: Game/Source/GDKTestGyms/AUptimeGameMode.h [hunk 2]
```diff
@@ -30,9 +31,13 @@ class GDKTESTGYMS_API AUptimeGameMode : public ABenchmarkGymGameModeBase
 	virtual void BuildExpectedActorCounts() override;
 	virtual void OnAnyWorkerFlagUpdated(const FString& FlagName, const FString& FlagValue) override;
 
+	UPROPERTY(EditAnywhere, NoClear, BlueprintReadOnly, Category = Classes)
+		TSubclassOf<AUptimeCrossServerBeacon> CrossServerClass;
+
 private:
 	TArray<FBlackboardValues> PlayerRunPoints;
 	TArray<FBlackboardValues> NPCRunPoints;
+	TArray<FBlackboardValues> CrossServerRunPoints;
```
Comment (jayprobable) [inline Game/Source/GDKTestGyms/AUptimeGameMode.h:40]: We should be able to remove these.. the beacons should be stationary

File: Game/Source/GDKTestGyms/UptimeCrossServerBeacon.h [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+#include "NFRConstants.h"
+#include "GameFramework/Character.h"
+#include "UptimeCrossServerBeacon.generated.h"
+
+DECLARE_LOG_CATEGORY_EXTERN(LogUptimeCrossServerBeacon, Log, All);
+
+UCLASS()
+class AUptimeCrossServerBeacon :
+	public ACharacter
```
Comment (jayprobable) [inline Game/Source/GDKTestGyms/UptimeCrossServerBeacon.h:13]: This shouldn't be a character it just needs to derive from actor

**Review by jayprobable — CHANGES_REQUESTED** [review]

File: Game/Source/GDKTestGyms/UptimeCrossServerBeacon.h [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+#include "NFRConstants.h"
+#include "GameFramework/Character.h"
+#include "UptimeCrossServerBeacon.generated.h"
+
+DECLARE_LOG_CATEGORY_EXTERN(LogUptimeCrossServerBeacon, Log, All);
+
+UCLASS()
+class AUptimeCrossServerBeacon :
+	public AActor
```
Comment (jeffxu-improbable) [inline Game/Source/GDKTestGyms/UptimeCrossServerBeacon.h:13]: `class AUptimeCrossServerBeacon : public AActor`
I'd like to put these in the same line instead.

**Review by jeffxu-improbable — APPROVED** [review]
Other than a few minor code style issues, everything else is find now.

**Review by jayprobable — APPROVED** [review]

**Pull request merged**